lib/sles4sap/sap_deployment_automation_framework/inventory_tools.pm

```diff
@@ -296,3 +298,46 @@ sub translate_hosts_data {
     } } keys(%hosts_data);
     return %result;
 }
+
+
+=head2 sdaf_create_instances
+
+    sdaf_create_instances(inventory_content=>HASHREF);
+
+Creates and returns  B<$instances> class which is a main component of F<lib/sles4sap_publiccloud.pm> and
+general public cloud libraries F</lib/publiccloud/*>.
+Check SDAF inventory file example in B<SYNOPSIS>
+
+=over
+
+=item * B<inventory_content> Referenced content of the SDAF inventory yaml file
+
+=item * B<sut_ssh_key_path> Path to private key file allowing SSH connection to SUT
+
+=back
+=cut
+
+sub sdaf_create_instances {
+    my (%args) = @_;
+    my @instances;
+
+    for my $instance_type (keys(%{$args{inventory_content}})) {
+        my $hosts = $args{inventory_content}->{$instance_type}{hosts};
+        for my $physical_host (keys %$hosts) {
+            my $instance = publiccloud::instance->new(
+                public_ip => $hosts->{$physical_host}->{ansible_host},
+                instance_id => $physical_host,
+                username => $hosts->{$physical_host}->{ansible_user},
+                ssh_key => $args{sut_ssh_key_path},
+                # Provider does not seem to be needed for SDAF as SDAF does AZ authentication differently
+                # Calling it causes a lot of troubles like creating ssh keys and .ssh/config which breaks SSH connections
+                provider => 'dummy',
+                region => get_required_var('PUBLIC_CLOUD_REGION')
+            );
+            push(@instances, $instance);
+        }
+    }
+
+    publiccloud::instances::set_instances(@instances);
```

Comment:
where is setting it? Is this function both returning reference to caller and store it somewhere down in PC?

Reply:
I was mostly following how it is done in qe-sap-deployment: 
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/033ea615cdbbf628b2a59c35201578d296fb92b1/lib/sles4sap_publiccloud.pm#L834

I am not sure if this is entirely required but it seems to only update global variable @instances:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/033ea615cdbbf628b2a59c35201578d296fb92b1/lib/publiccloud/instances.pm#L17

I added it as I am not sure if PC lib uses this global variable underneath somewhere ... sometime...

Reply:
I see, and maybe same suggestion is valid for that test too: move this `publiccloud::instance::set_instances` to https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/033ea615cdbbf628b2a59c35201578d296fb92b1/tests/sles4sap/publiccloud/qesap_terraform.pm#L217


My point is to avoid manipulating and storing data in the same place.

Reply:
I did follow the suggestion below. It is now in the test module instead. 
Thanks.